tools/build-tool/data/android/integrated-services/admob/app/src/main/java/net/sourceforge/castleengine/ServiceAdMob.java

```diff
@@ -42,37 +56,181 @@ private void initialize(String bannerUnitId, String interstitialUnitId, String[]
         initialized = true;
         mBannerUnitId = bannerUnitId;
         mInterstitialUnitId = interstitialUnitId;
+        mRewardedUnitId = rewardedUnitId;
         testDeviceIds = aTestDeviceIds;
         interstitialInitialize();
+        rewardedInitialize();
         logInfo(CATEGORY, "AdMob initialized");
     }
 
-    private void fullScreenAdClosed(boolean watched)
+    private void fullScreenAdClosed(TAdWatchStatus watchedStatus)
     {
-        messageSend(new String[]{"ads-admob-full-screen-ad-closed", booleanToString(watched)});
+        messageSend(new String[]{"ads-admob-full-screen-ad-closed", Integer.toString(watchedStatus.ordinal()) });
     }
 
     private void interstitialInitialize()
     {
+        if (mInterstitialUnitId == "") 
```

Comment:
This should use `equals` method, like `mInterstitialUnitId.equals("")`, see my other comment.